mlte/validation/validated_spec.py

```diff
@@ -0,0 +1,111 @@
+"""
+mlte/spec/validated_spec.py
+
+ValidatedSpec class implementation.
+"""
+
+from __future__ import annotations
+
+from typing import Dict, cast
+
+from mlte.artifact.artifact import Artifact
+from mlte.artifact.model import ArtifactModel
+from mlte.spec.model import SpecModel
+from mlte.spec.spec import Spec
+from mlte.validation.result import Result
+
+# -----------------------------------------------------------------------------
+# ValidatedSpec
+# -----------------------------------------------------------------------------
+
+
+class ValidatedSpec(Artifact):
+    """
+    ValidatedSpec represents a spec with validated results.
+    """
+
+    def __init__(self, spec: Spec, results: Dict[str, Dict[str, Result]]):
+        """
+        Initialize a ValidatedSpec instance.
+
+        :param spec: The Spec
+        :param results: The validation Results for the Spec
+        """
+
+        self.spec = spec
+        """The spec that we validated."""
+
+        self.results = results
+        """The validation results for the spec, by property."""
+
+        # Check that all conditions have results.
+        for property, conditions in self.spec.properties.items():
+            for measurement_id in conditions.keys():
+                if (
+                    property.name not in self.results
+                    or measurement_id not in self.results[property.name]
+                ):
+                    raise RuntimeError(
+                        f"Id '{measurement_id}' does not have a result."
+                    )
+
+    def to_model(self) -> ArtifactModel:
+        """
+        Generates a model representation of the ValidatedSpec.
+
+        :return: The serialized model
+        """
+        model = self.spec.to_model()
+        model.header.identifier = f"{model.header.identifier}.validated"
+
+        # Add results to main model.
+        spec_model: SpecModel = cast(SpecModel, model.body)
```

Comment:
This is the only other thing I am unsure about. If we reuse the `Spec` artifact type in the `ValidatedSpec` model, won't it be hard to distinguish between these two types, for instance in the frontend? Would it be difficult to simply reuse the model components from `SpecModel` in a `ValidatedSpecModel` type that has its own artifact type?

Reply:
Yes, it would be difficult. I was unsure about this structure too. If there are benefits, like the one you mention, I can create a simple ValidatedSpecModel that wraps or is similar to the Spec one to ensure the artifacts are treated differently, sure. I'll work on this.